src/repositories/request-repository.ts

```diff
@@ -21,15 +21,35 @@ export const FAILURE_RETRY_WINDOW = 1000 * 60 * 60 * 48 // 48H
 const REPEATED_READ_SERIALIZATION_ERROR = '40001'
 export const TABLE_NAME = 'request'
 
-const countRetryMetrics = (requests: Request[], anchoringDeadline: Date): void => {
-  const expired = requests.filter((request) => request.createdAt < anchoringDeadline)
-  if (expired.length > 0) Metrics.count(METRIC_NAMES.RETRY_EXPIRING, expired.length)
-
-  const processing = requests.filter((request) => request.status === RequestStatus.PROCESSING)
-  if (processing.length > 0) Metrics.count(METRIC_NAMES.RETRY_PROCESSING, processing.length)
+/**
+ * Records statistics about the set of requests
+ * Groups by EXPIRED (if the time is past the deadline), PROCESSING, and FAILED
+ * 
+ * Will record the total count, the mean time since createdAt, and the max time since createdAt
+ * for each group
+ *
+ * @param requests
+ * @param anchoringDeadline
+ * @returns
+ */
+const recordAnchorRequestMetrics = (requests: Request[], anchoringDeadline: Date): void => {
+  const expired = new TimeableMetric(SinceField.CREATED_AT)
```

Comment:
I'm not sure if this is actually what we want to track for expired, processing, and failed. I would think we would want the time elapsed since we last tried them, but I guess thats the timeouts. But what would we do with these timing metrics?

Reply:
oh - I thought this is what you had spec'd in the ticket, did I do it wrong?   (https://linear.app/3boxlabs/issue/CDB-1982/add-metrics-for-status-changes)

I think we want to know overall how long it takes for a request to get anchored so we can alert if it is too long?

Reply:
This one would be addressing the first of the three metrics outlined in the ticket. For these retried ones we are recording the "time elapsed since they were created every time" we retry it. I'm wondering what we would use that metric for.

Reply:
ok i think made all the changes we discussed!